x-pack/plugins/security/server/session_management/session_index.ts

```diff
@@ -484,24 +553,30 @@ export class SessionIndex {
       });
     }
 
-    try {
-      const { body: response } = await this.options.elasticsearchClient.deleteByQuery(
-        {
-          index: this.indexName,
-          refresh: true,
-          body: { query: { bool: { should: deleteQueries } } },
-        },
-        { ignore: [409, 404] }
-      );
+    // Create point in time snapshot to paginate through sessions
+    const { body: pitResponse } = await this.options.elasticsearchClient.openPointInTime({
+      index: this.indexName,
+      keep_alive: SESSION_INDEX_CLEANUP_KEEP_ALIVE,
+    });
```

Comment:
You need to close the PIT after you are finished searching (`elasticsearchClient.closePointInTime`)